internal/venice-client-common/src/main/java/com/linkedin/venice/schema/SchemaReader.java

```diff
@@ -17,5 +17,14 @@ public interface SchemaReader extends Closeable {
 
   Schema getLatestValueSchema();
 
+  /**
+   * Get the latest value schema id. This may be different from the value schema with the largest id if the superset
+   * schema is not the value schema with the largest id
+   */
   Integer getLatestValueSchemaId();
+
+  /**
+   * Get the largest used value schema id
+   */
+  int getMaxValueSchemaId();
```

Comment:
As per internal discussion, I will try to push down the logic to fetch derived schemas down into the `SchemaReader` interface. Adding this comment to prevent merging this PR accidentally